lint/settings.py

```diff
@@ -126,13 +131,91 @@ def validate_project_settings(filename):
     except ValueError:
         return False
 
-    settings = obj.get('SublimeLinter', {})
-    if len(settings.keys()) >= 2:
+    if 'SublimeLinter' in obj:
+        print_deprecation_message(obj.get('SublimeLinter', {}))
+        return False
+
+    settings = obj.get('settings', {})
+    if not settings:
+        util.clear_message()
+        return True
+
+    sl_settings = {
+        key: value
+        for key, value in settings.items()
+        if key.startswith('SublimeLinter.')
+    }
+    if not sl_settings:
+        util.clear_message()
+        return True
+
+    invalid_top_level_keys = [
+        key
+        for key in sl_settings
+        if not key.startswith('SublimeLinter.linters.')
+    ]
+    if invalid_top_level_keys:
+        logger.error(
+            "Invalid settings in '{}':\n"
+            "Only 'SublimeLinter.linters.*' keys are allowed. "
+            "Got {}.".format(
+                filename,
+                ', '.join(map(repr, invalid_top_level_keys))
+            )
+        )
+        return False
+
+    invalid_deep_keys = [
+        key
+        for key in sl_settings
+        if len(key.rstrip('.').split('.')) < 4
+    ]
+    if invalid_deep_keys:
         logger.error(
             "Invalid settings in '{}':\n"
-            "Only the key 'linters' is allowed. "
-            "Got {}.".format(filename, ', '.join(map(repr, settings.keys())))
+            "{} {} too short.".format(
+                filename,
+                ', '.join(map(repr, invalid_deep_keys)),
+                'are' if len(invalid_deep_keys) > 1 else 'is'
+            )
         )
         return False
 
-    return validate_settings([(filename, settings)])
+    deep_settings = {}
+    for key, value in sl_settings.items():
+        _, *parts = key.split('.')
+        edge = deep_settings
+        for part in parts[:-1]:
+            edge = edge.setdefault(part, {})
+
+        edge[parts[-1]] = value
+
+    return validate_settings([(filename, deep_settings)], flat=True)
+
+
+def print_deprecation_message(settings):
+    import json
+
+    message = """
+    Project settings for SublimeLinter have a new, flat format following the
+    Sublime standard. The old-style has been deprecated.
```

Comment:
Suggestion:

"following Sublime Text conventions. The old format has been deprecated, please use this format instead:"